### PR TITLE
feat: Query equipment-category/weapons by property

### PR DESCRIPTION
**Pull request by ThomasGCarpenter**
## What does this do?
On the equipment-categories route, one can now filter weapons by weapon properties. 

## How was it tested?
Manually tested various http requests for equipment-categories on Postman successfully. Followed swagger README instructions to set up local development environment which ran successfully after adding new openapi documentation. 

## Is there a Github issue this is resolving?
Yes, this resolves issue #232 

## Was any impacted documentation updated to reflect this change?
Yes, I updated the swagger documentation according to the guidelines found in the swagger README. 

## Here's a fun image for your troubles
random photo - update me

### CODE REVIEW

**Comment by bagelbits** [comment]
@ThomasGCarpenter Can you look into why your tests are failing?

**Comment by ThomasGCarpenter** [comment]
All the tests have now passed @bagelbits

File: src/controllers/api/equipmentCategoryController.ts [hunk 1]
```diff
@@ -1,4 +1,41 @@
...
 import SimpleController from '../simpleController.js';
+import { Request, Response, NextFunction } from 'express';
+import { APIReference } from '../../models/common/types.js';
 
-export default new SimpleController(EquipmentCategory);
+export const index = async (req: Request, res: Response, next: NextFunction) => {
+  const instantiatedSimpleController = new SimpleController(EquipmentCategory);
+  const indexResult = instantiatedSimpleController.index(req, res, next);
+  return indexResult;
+};
```
Comment (bagelbits) [inline src/controllers/api/equipmentCategoryController.ts:11]:
```suggestion
const simpleController = new SimpleController(Subclass);

export const index = async (req: Request, res: Response, next: NextFunction) =>
  simpleController.index(req, res, next);
```

File: src/controllers/api/equipmentCategoryController.ts [hunk 2]
```diff
@@ -1,4 +1,41 @@
...
+
+export const show = async (req: Request, res: Response, next: NextFunction) => {
+  try {
+    const requestedEquipmentCategory = await EquipmentCategory.findOne({
+      index: req.params.index,
+    });
+    if (!requestedEquipmentCategory) return next();
```
Comment (bagelbits) [inline src/controllers/api/equipmentCategoryController.ts:18]:
```suggestion
    if (!requestedEquipmentCategory) return res.status(404).json({ error: 'Not found' });;
```

**Comment by bagelbits** [comment]
Sorry. I think I misunderstood what this PR was trying to do. You're altering the `/api/equipment_category/${index}` route to filter the weapons that are returned with it by `property`?
You could maybe do something with `/api/equipment` route that would make a bit more sense. Alternatively, you can easily do this query with the GraphQL endpoint.

**Pull request closed without merging**